src/kohlrahbi/table_header.py

```diff
@@ -0,0 +1,170 @@
+
+            collector: Dict[str, Dict[str, str | int]] = {
+                pruefidentifikator: {
+                    HeaderSection.BESCHREIBUNG.value: "",
+                    HeaderSection.KOMMUNIKATION_VON.value: "",
+                    "tabstop_position": tab_stop,
+                }
+                for pruefidentifikator, tab_stop in zip(splitted_text, current_tabstop_positions)
+            }
+
+            if not collector:
+                raise ValueError("collector should not be empty")
+
+            return collector
+
+        if not row_cell.paragraphs[-1].text.startswith("Prüfidentifikator"):
+            raise ValueError("The last paragraph should start with 'Prüfidentifikator'")
+
+        collector = initialize_collector(paragraph=row_cell.paragraphs[-1])
+
+        section_type: HeaderSection
+
+        for paragraph in row_cell.paragraphs:
+            splitted_text = paragraph.text.split("\t")
+            text_prefix = splitted_text[0]
+
+            if text_prefix == "Prüfidentifikator":
+                continue
+
+            if text_prefix in ("Beschreibung", "Kommunikation von"):
+                initial_tabstop_positions = get_tabstop_positions(paragraph=paragraph)
+                tabstop_mapper = dict(zip(initial_tabstop_positions, collector.keys()))
+                section_type = HeaderSection[text_prefix.replace(" ", "_").upper()]
+                splitted_text.pop(0)
+                for pruefi, text in zip(collector.keys(), splitted_text):
+                    collector[pruefi][section_type.value] = text + " "
+            else:
+                if "" in splitted_text and len(splitted_text) > len(collector.keys()):
+                    splitted_text.remove("")
+
+                for tabstop_position, text in zip(initial_tabstop_positions, splitted_text):
+                    pruefi = tabstop_mapper[tabstop_position]
+                    collector[pruefi][section_type.value] += text + " "
+
+        def ensure_single_space_between_words(text: str) -> str:
+            return " ".join(text.split())
+
+        pruefi_meta_data = [
+            PruefiMetaData(
+                pruefidentifikator=pruefi,
+                communication_direction=ensure_single_space_between_words(
+                    #  The cast function is a no-op at runtime and doesn't perform any actual type conversion;
+                    #  it's only used for type checking purposes.
+                    cast(str, meta_data[HeaderSection.KOMMUNIKATION_VON.value])
+                ),
+                name=ensure_single_space_between_words(cast(str, meta_data[HeaderSection.BESCHREIBUNG.value])),
+            )
+            for pruefi, meta_data in collector.items()
+            if isinstance(meta_data[HeaderSection.KOMMUNIKATION_VON.value], str)
+        ]
+
+        return cls(pruefi_meta_data=pruefi_meta_data)
+
+    def get_pruefidentifikatoren(self) -> List[str]:
+        """
+        Get all Prüfidentifikatoren
```

Comment:
is the list sorted in anyway, maybe implicitly? are there duplicates?

Reply:
Fair point. I added a more detailed comment: e6a9aa58bfec03c4bb45322cdf0ba5ba8022bd99